### PR TITLE
[examples] Update manipulation_station meshcat interface.

### PR DESCRIPTION
**Pull request by svenevs**
Toward https://github.com/RobotLocomotion/drake/issues/14702, extracted parts from https://github.com/RobotLocomotion/drake/pull/16816.  Additionally, see [slack discussion here](https://drakedevelopers.slack.com/archives/C3YB3EV5W/p1652813084181929).

- Replace the deprecated pure python meshcat connection with the C++ interface.
- `args.meshcat` is now irrelevant, the meshcat visualizer is connected unconditionally.
- Add `-w` / `--open-window` to automatically open a browser to the meshcat server, argument options chosen to match meldis.

To test this PR, in one terminal run `bazel run //examples/manipulation_station:mock_station_simulation` to start the simulation.

In another terminal, two tests:

- `bazel run //examples/manipulation_station:joint_teleop -- --setup planar -w`: you will have the original `PlanarSceneGraphVisualizer` open, a new browser window with an orthographic meshcat view, and a slider panel display.  Confirm that the sliders control both visualizers.  `ctrl+c` out before the next test.
- `bazel run //examples/manipulation_station:joint_teleop -- --setup manipulation_class -w` will open a meshcat viewer and a slider panel display, there will be more sliders available than the planar version, confirm they manipulate the viewer.

~~**Note**: in order to run these commands, please locally apply or comment out the changeset from https://github.com/RobotLocomotion/drake/pull/17216 to get past the `NoneType` error.~~ Rebased.

---
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217)

### CODE REVIEW

**Review by svenevs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217)* status: 2 unresolved discussions, needs platform reviewer assigned, needs at least two assigned reviewers, missing label for release notes

___
*[`examples/manipulation_station/joint_teleop.py` line 89 at r1](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IXyaz-Cx9hjiFOWuF:-N2IXyaz-Cx9hjiFOWuG:b-gqy6hz) ([raw file](https://github.com/robotlocomotion/drake/blob/b1e3bc0bb1d7c5c5dbc85ef37e1fd0ef6b6634ac/examples/manipulation_station/joint_teleop.py#L89)):*
> ```Python
>         DrakeVisualizer.AddToBuilder(builder, geometry_query_port)
>         meshcat = Meshcat()
>         meshcat_params = MeshcatVisualizerParams()
> ```

It's not entirely clear to me if we actually need to keep these around, AFAICT if not provided they'll get defaulted.
___
*[`examples/manipulation_station/joint_teleop.py` line 99 at r1](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IYBQ1-VOcnIE3fl-M:-N2IYBQ1-VOcnIE3fl-N:bm7epnd) ([raw file](https://github.com/robotlocomotion/drake/blob/b1e3bc0bb1d7c5c5dbc85ef37e1fd0ef6b6634ac/examples/manipulation_station/joint_teleop.py#L99)):*
> ```Python
>             meshcat.Set2dRenderMode()
>             pyplot_visualizer = ConnectPlanarSceneGraphVisualizer(
>                 builder, station.get_scene_graph(), geometry_query_port)
> ```

Originally this was getting deleted, do we want to keep it or get rid of it (the meshcat viewer is way sleeker, but users have to know to `-w` it otherwise nothing shows up).

**Review by svenevs — COMMENTED** [review]
+(release notes: none) the data for this example is installed, but `joint_teleop` itself is not.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217#-:-N2IZoji9fZ_98G6xbE6:b-tu0uxw)* status: 2 unresolved discussions, needs platform reviewer assigned, needs at least two assigned reviewers

**Review by svenevs — COMMENTED** [review]
+@BetsyMcPhail for feature review please.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217#-:-N2IkmPa-gE2kGd9DC3g:bs3ynz8)* status: 2 unresolved discussions, LGTM missing from assignee BetsyMcPhail, needs platform reviewer assigned, needs at least two assigned reviewers (waiting on @BetsyMcPhail)

**Review by BetsyMcPhail — COMMENTED** [review]
I confirmed that I'm able to follow the testing instructions on my local machine.

:lgtm: 

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217#-:-N2InApX-D2fVMs2Gd_x:b-khj3x0)* status: 2 unresolved discussions, needs platform reviewer assigned, needs at least two assigned reviewers (waiting on @BetsyMcPhail)

**Review by BetsyMcPhail — COMMENTED** [review]
+@rpoyner-tri for platform review please. I believe you have some familiarity with this file, if you could also help with Stephen's outstanding questions it would be appreciated.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217#-:-N2InU7_9OHdUluoSFRp:bgh6w46)* status: 2 unresolved discussions, LGTM missing from assignee rpoyner-tri(platform) (waiting on @BetsyMcPhail and @rpoyner-tri)

**Review by svenevs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217)* status: 2 unresolved discussions, LGTM missing from assignee rpoyner-tri(platform) (waiting on @BetsyMcPhail and @rpoyner-tri)

___
*[`examples/manipulation_station/joint_teleop.py` line 105 at r2](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IoJ-MAQSmPMl2Eqbd:-N2IoJ-MAQSmPMl2Eqbe:bobwxco) ([raw file](https://github.com/robotlocomotion/drake/blob/0b3c590b5029ac46b6d581c12a0085ac11bacc9c/examples/manipulation_station/joint_teleop.py#L105)):*
> ```Python
>             webbrowser.open(url=url, new=args.browser_new)
> 
>     teleop = builder.AddSystem(JointSliders(station.get_controller_plant(),
> ```

BTW There is a related problem https://github.com/RobotLocomotion/drake/pull/16816#pullrequestreview-944601516

Previously and in this current PR `meshcat` is **not** created when `args.hardware` is used (not something I can test).  On slack we discussed unconditionally adding it, but in the current format the visualizers are only added when `args.hardware` is not provided.  However, `JointSliders` is ultimately what is getting removed in favor of the meschat sliders.  It's not clear to me what to do when `args.hardware` is given, but the same comes up with the buttons.

I don't think this necessarily needs to block this PR, but if you have an idea of what the right thing to do is when we're actually working with the robot that would be helpful.  Should this PR instead just truly always be adding `meshcat` (even when `args.hardware` is provided) AKA dedent it out of the `if`?

Basically, we're not sure how to approach `args.hardware`, we could fix it here or in a followup that actually does the replacement.



_Code quote:_
```Python
    teleop = builder.AddSystem(JointSliders(station.get_controller_plant(),
                                            length=800))
```

**Review by rpoyner-tri — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217#-:-N2JJyk65EVua1RcdV4A:bnfp4nl)* status: 2 unresolved discussions (waiting on @BetsyMcPhail, @rpoyner-tri, and @svenevs)

___
*[`examples/manipulation_station/joint_teleop.py` line 89 at r1](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IXyaz-Cx9hjiFOWuF:-N2JDupy0eg8OL6p8XIA:bq1qnd6) ([raw file](https://github.com/robotlocomotion/drake/blob/b1e3bc0bb1d7c5c5dbc85ef37e1fd0ef6b6634ac/examples/manipulation_station/joint_teleop.py#L89)):*
<details><summary><i>Previously, svenevs (Stephen McDowell) wrote…</i></summary><blockquote>

It's not entirely clear to me if we actually need to keep these around, AFAICT if not provided they'll get defaulted.
</blockquote></details>

You can avoid mentioning params at all in this file -- they will get defaulted.
___
*[`examples/manipulation_station/joint_teleop.py` line 20 at r2](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2JIsCA3XZo6B8oeoi1:-N2JIsCA3XZo6B8oeoi2:b2uwjd7) ([raw file](https://github.com/robotlocomotion/drake/blob/0b3c590b5029ac46b6d581c12a0085ac11bacc9c/examples/manipulation_station/joint_teleop.py#L20)):*
> ```Python
> from pydrake.systems.framework import DiagramBuilder
> from pydrake.systems.analysis import Simulator
> from pydrake.geometry import Meshcat, MeshcatVisualizerParams, \
> ```

BTW the import of params can go away if the use of params below also goes away.
___
*[`examples/manipulation_station/joint_teleop.py` line 105 at r2](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IoJ-MAQSmPMl2Eqbd:-N2JIXrl7gMi5uvdCd6n:b-lt170s) ([raw file](https://github.com/robotlocomotion/drake/blob/0b3c590b5029ac46b6d581c12a0085ac11bacc9c/examples/manipulation_station/joint_teleop.py#L105)):*
<details><summary><i>Previously, svenevs (Stephen McDowell) wrote…</i></summary><blockquote>

BTW There is a related problem https://github.com/RobotLocomotion/drake/pull/16816#pullrequestreview-944601516

Previously and in this current PR `meshcat` is **not** created when `args.hardware` is used (not something I can test).  On slack we discussed unconditionally adding it, but in the current format the visualizers are only added when `args.hardware` is not provided.  However, `JointSliders` is ultimately what is getting removed in favor of the meschat sliders.  It's not clear to me what to do when `args.hardware` is given, but the same comes up with the buttons.

I don't think this necessarily needs to block this PR, but if you have an idea of what the right thing to do is when we're actually working with the robot that would be helpful.  Should this PR instead just truly always be adding `meshcat` (even when `args.hardware` is provided) AKA dedent it out of the `if`?

Basically, we're not sure how to approach `args.hardware`, we could fix it here or in a followup that actually does the replacement.
</blockquote></details>

I don't have ready answers to these questions, and my on-call platform review time is evaporating. I think the changes to move to meshcat-cpp are fine, and perhaps the rest should be deferred to a separate step.

**Review by rpoyner-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217)* status: 2 unresolved discussions (waiting on @svenevs)

___
*[`examples/manipulation_station/joint_teleop.py` line 99 at r1](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IYBQ1-VOcnIE3fl-M:-N2JLe_52LLklzjms4EE:b-8ox2oy) ([raw file](https://github.com/robotlocomotion/drake/blob/b1e3bc0bb1d7c5c5dbc85ef37e1fd0ef6b6634ac/examples/manipulation_station/joint_teleop.py#L99)):*
<details><summary><i>Previously, svenevs (Stephen McDowell) wrote…</i></summary><blockquote>

Originally this was getting deleted, do we want to keep it or get rid of it (the meshcat viewer is way sleeker, but users have to know to `-w` it otherwise nothing shows up).
</blockquote></details>

This seems like a useful discussion, but one that could be separated from the current PR re: porting to c++ meshcat.

**Review by rpoyner-tri — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217)* status: 2 unresolved discussions (waiting on @svenevs)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2JOCoJ6GtWdUvDENop:-N2JOCoK0sC8XqJTV3oB:b-o5dfj) (no related file):*
A bit of reviewable.io technique: it's fine and useful to open discussions on your own PR, but the "Blocking" discussion state is probably the least useful to use as the PR author. Better to choose one of "Informing", "Discussing", or "Working". Working is preferable if it is accurate, because it removes the red alert color from other reviewers' dashboards.

___

**Review by svenevs — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/17217)* status: :shipit: complete! all discussions resolved, LGTM from assignees rpoyner-tri(platform),BetsyMcPhail (waiting on @rpoyner-tri)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2JOCoJ6GtWdUvDENop:-N2MQV4SElikcIdF1Pcs:b3a2qd6) (no related file):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

A bit of reviewable.io technique: it's fine and useful to open discussions on your own PR, but the "Blocking" discussion state is probably the least useful to use as the PR author. Better to choose one of "Informing", "Discussing", or "Working". Working is preferable if it is accurate, because it removes the red alert color from other reviewers' dashboards.
</blockquote></details>

Ok thank you for the feedback, I will use `Working` for blocking comments that I need help with in the future.

___
___
*[`examples/manipulation_station/joint_teleop.py` line 89 at r1](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IXyaz-Cx9hjiFOWuF:-N2MQfqE8uVREuWTyqrT:brg1kxl) ([raw file](https://github.com/robotlocomotion/drake/blob/b1e3bc0bb1d7c5c5dbc85ef37e1fd0ef6b6634ac/examples/manipulation_station/joint_teleop.py#L89)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

You can avoid mentioning params at all in this file -- they will get defaulted.
</blockquote></details>

Done, params are defaulted by `AddToBuilder` now.
___
*[`examples/manipulation_station/joint_teleop.py` line 99 at r1](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IYBQ1-VOcnIE3fl-M:-N2MQkVt8JeOicpWcFiO:b274byt) ([raw file](https://github.com/robotlocomotion/drake/blob/b1e3bc0bb1d7c5c5dbc85ef37e1fd0ef6b6634ac/examples/manipulation_station/joint_teleop.py#L99)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

This seems like a useful discussion, but one that could be separated from the current PR re: porting to c++ meshcat.
</blockquote></details>

OK I will followup on that in the future.
___
*[`examples/manipulation_station/joint_teleop.py` line 20 at r2](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2JIsCA3XZo6B8oeoi1:-N2MQdIF3vpbQljaVyNr:b957j8l) ([raw file](https://github.com/robotlocomotion/drake/blob/0b3c590b5029ac46b6d581c12a0085ac11bacc9c/examples/manipulation_station/joint_teleop.py#L20)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

BTW the import of params can go away if the use of params below also goes away.
</blockquote></details>

Done, good eye.
___
*[`examples/manipulation_station/joint_teleop.py` line 105 at r2](https://reviewable.io/reviews/robotlocomotion/drake/17217#-N2IoJ-MAQSmPMl2Eqbd:-N2MQoR287NgcmOBAmHl:bv9rx5i) ([raw file](https://github.com/robotlocomotion/drake/blob/0b3c590b5029ac46b6d581c12a0085ac11bacc9c/examples/manipulation_station/joint_teleop.py#L105)):*
<details><summary><i>Previously, rpoyner-tri (Rick Poyner (rico)) wrote…</i></summary><blockquote>

I don't have ready answers to these questions, and my on-call platform review time is evaporating. I think the changes to move to meshcat-cpp are fine, and perhaps the rest should be deferred to a separate step.
</blockquote></details>

OK no problem, this will get resolved when adding the sliders / button.
